app/jobs/autoscaling.py

```diff
@@ -260,227 +61,313 @@ def get_instance_tags():
     tag_dict = json.loads(aws_command(cmd))
     return tag_dict["Tags"]
 
-
-def instances_in(asg):
-    return [item["InstanceId"] for item in asg["Instances"]]
-
-
-def find_tags_for_instances_in(asg, tag_list):
-    instance_ids = instances_in(asg)
-    return [tag for tag in tag_list if tag["ResourceId"] in instance_ids]
-
-
-def add_draining_tag(instance_ids, draining_tag):
-    cmd = "aws ec2 create-tags --resources {list_instances} --tags Key={draining_tag},Value={timestamp}"
-    cmd = cmd.format(list_instances=' '.join(instance_ids), draining_tag=draining_tag, timestamp=unixtime_now())
+def delete_tags_from_instances(instance_ids, tag_keys):
+    if not instance_ids or not tag_keys:
+        return
+    cmd = "aws ec2 delete-tags --resources {list_instances} --tags {list_tags}"
+    cmd = cmd.format(list_instances=' '.join(instance_ids), list_tags=' '.join(['Key='+tag for tag in tag_keys]))
     aws_command(cmd)
 
+def get_asg_list():
+    asg_json = aws_command("aws autoscaling describe-auto-scaling-groups")
+    asg_list = json.loads(asg_json).get('AutoScalingGroups', [])
+    return asg_list
 
-def remove_draining_tag(instance_ids, draining_tag):
-    delete_tags_from_instances(instance_ids, [draining_tag])
-
+def required_capacity(num_jobs):
+    if num_jobs == 0:
+        result = exactly(0)
+    elif num_jobs == 1:
+        # Often a test job or automatic benchmark job
+        result = exactly(4)
+    elif 1 < num_jobs <= 6:
+        result = exactly(8)
+    else:
+        # This means, as long as we're in this regime, a DesiredCapacity > 24 set manually by the operator
+        # will be respected and not scaled down until we exit this regime.
+        result = at_least(24)
+    return result
 
-def start_draining(asg_instance_name, draining_tag, num_instances):
-    instance_ids = instances_to_drain(asg_instance_name, draining_tag, num_instances)
-    if instance_ids:
-        print "Starting to drain the following instances: " + ",".join(instance_ids)
-        add_draining_tag(instance_ids, draining_tag)
+def autoscaling_update(my_num_jobs, my_environment="development",
+                       max_job_dispatch_lag_seconds=900, job_tag_prefix="RunningIDseqBatchJob_",
+                       job_tag_keep_alive_seconds=600, draining_tag="draining"):
```

Comment:
will these params ever be modified here?   if not, i'd just move them in the ASG constructor, so we don't have to pass them through

Reply:
Yes, they will be modified, in fact defaults are only for documentation. Maybe I should remove them. They're passed from the DAG template to (a) the autoscaler, (b) the idseq-dag code

Reply:
ok done